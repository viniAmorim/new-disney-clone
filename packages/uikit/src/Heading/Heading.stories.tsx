import React from 'react';

import { Story, Meta } from '@storybook/react';

import Heading, { HeadingProps } from '~/Heading';

export default {
  title: 'Typography/Heading',
  component: Heading,
} as Meta;

const Template: Story<HeadingProps> = (args) => <Heading {...args} />;

export const DisplayHeavy = Template.bind({});
DisplayHeavy.args = {
  children: 'Display',
  size: 'display',
};

export const HeadlineHeavy = Template.bind({});
HeadlineHeavy.args = {
  children: 'Headline',
  size: 'headline',
};

export const TitleHeavy = Template.bind({});
TitleHeavy.args = {
  children: 'Title',
  size: 'title',
};

export const SubheaderHeavy = Template.bind({});
SubheaderHeavy.args = {
  children: 'Subheader',
  size: 'subheader',
};

export const BodyHeavy = Template.bind({});
BodyHeavy.args = {
  children: 'Body',
  size: 'body',
};

export const SmallHeavy = Template.bind({});
SmallHeavy.args = {
  children: 'Small',
  size: 'small',
};
