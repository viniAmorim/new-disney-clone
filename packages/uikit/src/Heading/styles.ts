import styled, { css } from 'styled-components';

export type StyledHeadingProps = {
  size?: 'display' | 'headline' | 'title' | 'subheader' | 'body' | 'small';
};

const sizeModifier = {
  display: css`
    ${({ theme }) => theme.typography['display-heavy']}
    font-size: 36px;
    line-height: 48px;
  `,
  headline: css`
    ${({ theme }) => theme.typography['headline-heavy']}
    font-size: 24px;
    line-height: 32px;
  `,
  title: css`
    ${({ theme }) => theme.typography['title-heavy']}
    font-size: 20px;
    line-height: 28px;
  `,
  subheader: css`
    ${({ theme }) => theme.typography['subheader-heavy']}
    font-size: 16px;
    line-height: 24px;
  `,
  body: css`
    ${({ theme }) => theme.typography['body-heavy']}
    font-size: 14px;
    line-height: 20px;
  `,
  small: css`
    ${({ theme }) => theme.typography['small-heavy']}
    font-size: 12px;
    line-height: 16px;
  `,
};

export const Heading = styled.h1<StyledHeadingProps>`
  ${({ size = 'display' }) => sizeModifier[size]}
`;
