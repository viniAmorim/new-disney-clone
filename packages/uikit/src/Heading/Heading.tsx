import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface HeadingProps
  extends BaseHTMLAttributes<HTMLHeadingElement>,
    Styled.StyledHeadingProps {
  /**
   * Heading Content
   */
  children: ReactNode;
}

const Heading = ({ children, size, ...rest }: HeadingProps) => {
  return (
    <Styled.Heading size={size} {...rest}>
      {children}
    </Styled.Heading>
  );
};

export default Heading;
