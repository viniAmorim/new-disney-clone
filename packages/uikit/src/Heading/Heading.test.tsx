import React from 'react';

import { render } from '~/test/utils';

import Heading from './Heading';

describe('Heading', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Heading>Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveTextContent(/^Heading$/);

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text display', () => {
    const { getByText } = render(<Heading size="display">Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text headline', () => {
    const { getByText } = render(<Heading size="headline">Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text title', () => {
    const { getByText } = render(<Heading size="title">Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text subheader', () => {
    const { getByText } = render(<Heading size="subheader">Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text body', () => {
    const { getByText } = render(<Heading size="body">Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });

  it('should render a text small', () => {
    const { getByText } = render(<Heading size="small">Heading</Heading>);

    const textElement = getByText('Heading');

    expect(textElement).toBeInTheDocument();

    expect(textElement).toMatchSnapshot();
  });
});
