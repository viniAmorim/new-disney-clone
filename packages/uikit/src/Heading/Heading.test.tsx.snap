// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a text body 1`] = `
<h1
  class="sc-bdfBQB juIlDX"
>
  Heading
</h1>
`;

exports[`Heading should render a text display 1`] = `
<h1
  class="sc-bdfBQB ekhnXy"
>
  Heading
</h1>
`;

exports[`Heading should render a text headline 1`] = `
<h1
  class="sc-bdfBQB jThsMR"
>
  Heading
</h1>
`;

exports[`Heading should render a text small 1`] = `
<h1
  class="sc-bdfBQB dEUoSM"
>
  Heading
</h1>
`;

exports[`Heading should render a text subheader 1`] = `
<h1
  class="sc-bdfBQB frtOOl"
>
  Heading
</h1>
`;

exports[`Heading should render a text title 1`] = `
<h1
  class="sc-bdfBQB gJxtUI"
>
  Heading
</h1>
`;

exports[`Heading should render correctly 1`] = `
<h1
  class="sc-bdfBQB ekhnXy"
>
  Heading
</h1>
`;
